docs/protocol-basics.rst

```diff
@@ -138,8 +136,60 @@ block public key.
 
 .. _status:
 
-Status
-------
+Status (protocol 1.5 and later)
+-------------------------------
+
+To calculate the `status` of a :ref:`script hash <script hashes>` (or
+address):
+
+1. Consider all transactions touching the script hash (both those spending
+from it, and those funding it), both confirmed and unconfirmed (in mempool).
+
+2. Order confirmed transactions by increasing height (and position in the
+block if there are more than one in a block).
+
+3. For mempool transactions, we define **height** to be ``-1`` if the
+transaction has at least one unconfirmed input, and ``0`` if all inputs are
+confirmed.
+
+4. Order mempool transactions by ``(-height, tx_hash)``, that is,
+``0`` height txs come before ``-1`` height txs, and secondarily the
+txid (in network byteorder) is used to arrive at a canonical ordering.
+
+5. For each confirmed tx, form a bytearray: ``tx_hash+height``, where:
+
+  * ``tx_hash`` is the 32-byte transaction hash in network byteorder
+    (reverse of human display endianness)
+
+  * ``height`` is the height of the block the tx is included in,
+    serialised as 8-bytes, little-endian, signed (two's complement)
```

Comment:
Why 8 bytes?  Height can never be more than 4 bytes, which is 4 billion blocks (thousands of years).

Reply:
Sure, 4 bytes should also be sufficient. The max is 2 billion then, as it should be signed, due to sometimes using `-1` as value in the mempool case.
I don't think there is any performance difference between 4 vs 8 bytes, as sha256 pads to multiples of 64 bytes.
As there is no difference, my thinking was to make it larger so as to avoid some shitcoiner with a crazy high block frequency complaining in the future.

I don't care much either way though; we can use 4 bytes if someone does care.

Reply:
On second thought, 4 bytes really should be enough: with 80 byte headers, when overflowing that, the headers itself would use up 160 GiB space.